torch/distributed/distributed_c10d.py

```diff
@@ -351,7 +351,7 @@ def init_process_group(backend,
         elif world_size != -1:
             url += "?world_size={}".format(world_size)
 
-        store, rank, world_size = next(rendezvous(url))
+        store, rank, world_size = next(rendezvous(url, timeout=timeout))
```

Comment:
We can still call `set_timeout` (or an accessor to stay Pythonic) on the store object right after this returns.

No need to do it only for the Gloo backend. Applies to both the Gloo and NCCL backends.